### PR TITLE
Attemps to  "meson: Deny shared build on MSVC compiler"

### PR DESCRIPTION
**Pull request by animalize**

### CODE REVIEW

File: lib/common/xxhash.c [hunk 1]
```diff
@@ -459,7 +459,7 @@ FORCE_INLINE_TEMPLATE U64 XXH64_endian_align(const void* input, size_t len, U64
 }
 
 
-XXH_PUBLIC_API unsigned long long XXH64 (const void* input, size_t len, unsigned long long seed)
+XXHASH_DLL_PUBLIC XXH_PUBLIC_API unsigned long long XXH64 (const void* input, size_t len, unsigned long long seed)
```
Comment (Cyan4973) [inline lib/common/xxhash.c:462]: Why do you want to make `XXH64` public ?

File: lib/compress/zstd_compress_internal.h [hunk 2]
```diff
@@ -1173,6 +1182,6 @@ size_t ZSTD_referenceExternalSequences(ZSTD_CCtx* cctx, rawSeq* seq, size_t nbSe
 
 /** ZSTD_cycleLog() :
  *  condition for correct operation : hashLog > 1 */
-U32 ZSTD_cycleLog(U32 hashLog, ZSTD_strategy strat);
+ZSTD_COMPRESS_DLL_PUBLIC U32 ZSTD_cycleLog(U32 hashLog, ZSTD_strategy strat);
```
Comment (Cyan4973) [inline lib/compress/zstd_compress_internal.h:1185]: Why should `ZSTD_cycleLog()` become public ?

File: lib/dictBuilder/zdict.c [hunk 3]
```diff
@@ -969,15 +969,16 @@ static size_t ZDICT_addEntropyTablesFromBuffer_advanced(
 }
 
 /* Hidden declaration for dbio.c */
-size_t ZDICT_trainFromBuffer_unsafe_legacy(
+ZDICTLIB_API size_t ZDICT_trainFromBuffer_unsafe_legacy(
```
Comment (Cyan4973) [inline lib/dictBuilder/zdict.c:972]: I'm surprised, normally it's enough for `ZDICTLIB_API` to be on the declaration side (`zdict.h`). It shouldn't be needed on the implementation side (`zdict.c`).

**Comment by animalize** [comment]
Only the last commit is from me, it makes the CI green.
This PR is test for CI, as I said in https://github.com/lzutao/zstd/pull/2, I will close this PR.
No need to review in this one, sorry I didn't say it clearly.
@lzutao

**Comment by animalize** [comment]
I will try to solve them later.

There is no need to work these days, I read some news and understand these days are very bad, rest more, these works can be done later.

**Comment by animalize** [comment]
@lzutao 
It seems all changes in `.c` `.h` files can be removed.

**Pull request closed without merging**